### PR TITLE
feat(global): add robust typing and registry helper

### PR DESCRIPTION
**Pull request by Picodes**

### CODE REVIEW

File: src/keepers/staking.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import { BigNumber, Contract, providers, Wallet } from 'ethers';
 
-import { ALL_TOKENS, AngleContractsStableType, CONTRACTS_ADDRESSES, Interfaces, StableTokens } from '../constants';
+import { ALL_TOKENS, CONTRACTS_ADDRESSES, registry, RewardsDistributor, RewardsDistributor__factory } from '../constants';
```
Comment (sogipec) [inline src/keepers/staking.ts:3]: Some lint issues here

File: scripts/validate.ts [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import { utils } from 'ethers';
```
Comment (greedythib) [inline scripts/validate.ts:1]: we could also use Joi schema for this script, it provides a great debugger, especially when running the ci

File: src/constants/contracts_addresses/polygon.ts [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+const addresses: ContractsRegistryType['1'] = {
+  ANGLE: '0x900F717EA076E1E7a484ad9DD2dB81CEEc60eBF1',
+  AngleRouter: '0x892bf71463Bd9fa57f3c2266aB74dbe1B96DECEa',
+  agEUR: {
+    AgToken: '0xE0B52e49357Fd4DAf2c15e02058DCE6BC0057db4',
+    bridges: {
+      LayerZero: '0x0c1EBBb61374dA1a8C57cB6681bF27178360d36F',
+    },
+    borrowCollaterals: {
+      MAI: {
```
Comment (greedythib) [inline src/constants/contracts_addresses/polygon.ts:12]: can you add the new collat `agstk-am3CRV` with the LP token address?

**Review by greedythib — APPROVED** [review]

File: scripts/validate.ts (hunk 2 above)
Comment (Picodes) [reply scripts/validate.ts:1]: Joi schema is only for json no ? Like here it does not add anything as it's already typed

**Pull request merged**